openedx/core/lib/completion.py

```diff
@@ -0,0 +1,25 @@
+"""Helper methods for completion."""
+
+from xblock.completable import XBlockCompletionMode
+
+
+def should_mark_block_completed_on_view(block):
```

Comment:
Nit: consider renaming this to something like `can_block_be_completed_on_view`.  Just because a block meets the conditions in the body of this function, doesn't mean it _should_ be marked complete yet - we still have to know if it's already been completed (as you check for below).

Reply:
Renamed to `can_block_be_completed_on_view()`.